internal/pkg/cli/app_delete.go

```diff
@@ -47,6 +47,28 @@ type deleteAppOpts struct {
 	projectEnvironments []*archer.Environment
 }
 
+func newDeleteAppOpts() (*deleteAppOpts, error) {
```

Comment:
I agree we should reuse the `newDeleteAppOpts()` function within the `BuildAppDeleteCmd()`. I don't want to include that as part of this PR since we'll also need to change the flag bindings to use viper to ensure we can properly handle an error returned from calling this function in the `RunE` function.